matrix/Matrix.hpp

```diff
@@ -274,28 +274,14 @@ class Matrix
 
     void write_string(char * buf, size_t n) const
     {
+        buf[0] = '\0'; // make an empty string to begin with (we need the '\0' for strlen to work)
```

Comment:
i tested the print on nuttx but buffer overflow is a good point and got lost on my todo list, I'm solving it now and will report